src/hotspot/cpu/aarch64/frame_aarch64.cpp

```diff
@@ -268,17 +268,18 @@ bool frame::safe_for_sender(JavaThread *thread) {
 void frame::patch_pc(Thread* thread, address pc) {
   assert(_cb == CodeCache::find_blob(pc), "unexpected pc");
   address* pc_addr = &(((address*) sp())[-1]);
+  address signing_sp = (((address*) sp())[-2]);
+  address signed_pc = pauth_sign_return_address(pc, (address)signing_sp);
   if (TracePcPatching) {
     tty->print_cr("patch_pc at address " INTPTR_FORMAT " [" INTPTR_FORMAT " -> " INTPTR_FORMAT "]",
-                  p2i(pc_addr), p2i(*pc_addr), p2i(pc));
+                  p2i(pc_addr), p2i(*pc_addr), p2i(signed_pc));
```

Comment:
Let's see both pc and signed pc here.

Reply:
> Let's see both pc and signed pc here, if they are different.

Reply:
Are you sure? At the moment with PAC we get:

patch_pc at address 0x0000fffff58edf98 [0x0068ffffed17b5fc -> 0x00abffffed17b7f8]

With both signed and unsigned you'd have:

patch_pc at address 0x0000fffff58edf98 [0x0068ffffed17b5fc (0x0000ffffed17b5fc) -> 0x00abffffed17b7f8 (0x0000ffffed17b7f8)]

I prefer the first - it's shorter and you can infer the address from the signed version. Happy to go with the longer version if you think the shorter version is confusing.

Reply:
You've been looking at PAC-signed addresses for a long time. 
Let's see "at address [prev true dest -> new true dest] [signed prev signed dest -> new signed dest]", but only show the signed dests if they're different. So it appears as 
`patch_pc at address 0x0000fffff58edf98 [0x0068ffffed17b5fc -> 0x00abffffed17b7f8] [signed 0x0000ffffed17b5fc ->  0x0000ffffed17b7f8]` .

Reply:
ok, that looks better than my longer version, I'll go with that